[Wiki] Updated "CI Environment' to latest (#3499)
Azure/ResourceModules: docs/wiki/The CI environment - Pipeline design.md

## Module pipeline inputs

Each module pipeline comes with the following runtime parameters (which are identical for GitHub & AzureDevOps):

- `'Branch'` dropdown: A dropdown to select the branch to run the pipeline from.

> - `'Execute static validation'` switch: Can be enabled or disabled and controls whether the static tests jobs are executed during the pipeline run.

Reviewer: ```suggestion
- `'Execute static validation'` switch: Can be enabled or disabled. It controls whether the static tests jobs are executed during the pipeline run.
```

>   > **Note:** This switch cannot be used to bypass the publishing requirements that both the static & deployment validation must be successful.

Reviewer: ```suggestion
  > **Note:** This switch cannot be used to bypass the publishing requirements where both static tests & deployment validation must be successful.
```

> - `'Execute deployment validation'` switch: Can be enabled or disabled and controls whether the deployment tests jobs are executed during the pipeline run.

Reviewer: ```suggestion
- `'Execute deployment validation'` switch: Can be enabled or disabled. It controls whether the deployment validation jobs are executed during the pipeline run.
```

> - `'Remove deployed module'` switch: Can be enabled or disabled and controls whether the test-deployed resources are removed after testing. It is enabled by default.

Reviewer: ```suggestion
- `'Remove deployed module'` switch: Can be enabled or disabled. It controls whether the test-deployed resources are removed after the deployment validation jobs. It is enabled by default.
```

>   > **Note:** This switch cannot be used to bypass the publishing requirements that both the static & deployment validation must be successful.

Reviewer: ```suggestion
  > **Note:** This switch cannot be used to bypass the publishing requirements, where both the static tests & deployment validation jobs must be successful.
```

> - `'Publish prerelease module'` switch: Can be enabled or disabled and allows publishing a prerelease version for the corresponding module when running the pipeline from a branch different than `main|master`. It is disabled by default. For further information of how the input is processed refer to the [Publishing](./The%20CI%20environment%20-%20Publishing) dedicated page.

Reviewer: ```suggestion
- `'Publish prerelease module'` switch: Can be enabled or disabled. It allows publishing a `pre-release` version for the corresponding module when running the pipeline from a different branch than `main | master`. It is disabled by default. For further information of how the input is processed refer to the [Publishing Guide](./The%20CI%20environment%20-%20Publishing) dedicated page.
```
